python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -2091,20 +2089,19 @@ def __setattr__(self, attr, value):
 
             # reduce duplicate values here, leave only duplicates for LCG,
             # where we can have replicas
-            uniqueValuesDict = []
-            uniqueValues = []
+            uniqueValuesDict = oDict()
```

Comment:
because the ordering isn't strongly preserved when using a dict as with a list and the test fails (randomly), hence the ordered dict to stop this happening.